### PR TITLE
fix: dispatches actions to update SearchContext from SearchBox

### PR DESCRIPTION
**Pull request by chavesj**
## Description

- dispatches appropriate actions from `SearchBox`'s `handleSubmit` & `handleClear` methods
- adds renderWithSearchContext test util, updates SearchBox tests

## Ticket link

[ENT-4382](https://openedx.atlassian.net/browse/ENT-4382)

## Testing Instructions

I tested these changes within the Enterprise Learner Portal in a way that Algolia setup for my devstack was not required.

### Update Learner Portal webpack config

Within your local copy of frontend-app-learner-portal-enterprise add a module.config.js file with the following contents:
```
module.exports = {
  localModules: [
    { moduleName: '@edx/frontend-enterprise', dir: '../frontend-enterprise', dist: 'src' },
  ],
};
```
This will force webpack to use your local copy of frontend-enterprise when building the learner portal (note the value for `dir` within the module object and update if needed).

### Use React Developer Tools to inspect SearchData

Start the learner portal and navigate to `/<customerSlug>/search`. Open React Developer Tools to the _Components_ tab and select the `SearchData` component. Take note of `Reducer` and `History` in the _hooks_ pane to the right. First with the master branch checked out and then with this branch checked out, observe the values in the hooks pane after submitting a search and clearing it. 

On the master branch, `History` is correctly updated after clearing the search but the search query remains in the `Reducer`. With this branch checked out, the search query is correctly cleared from the `Reducer` and the `History` is still correctly updated.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,14 +71,6 @@
...
     "lodash.debounce": "^4.0.8"
   },
-  "jest": {
-    "collectCoverageFrom": [
-      "src/**/*.{js,jsx}",
-      "!src/index.js",
-      "!**/node_modules/**",
-      "!**/tests/**"
-    ]
-  },
```
Comment (chavesj) [inline package.json:81]: jest config not needed, config from frontend-build is used

File: src/utils/tests.jsx [hunk 2]
```diff
@@ -25,28 +24,8 @@ export function renderWithRouter(
...
-    error: null,
-    isMarkCourseCompleteSuccess: false,
-    isMoveToInProgressSuccess: false,
-  },
-  offers: {
-    loading: false,
-    offersCount: 0,
-    offers: [],
-  },
-};
```
Comment (chavesj) [inline src/utils/tests.jsx:52]: removes unused test utils

File: src/utils/tests.jsx [hunk 3]
```diff
@@ -25,28 +24,8 @@ export function renderWithRouter(
   };
 }
 
-/* eslint-disable react/prop-types */
-export const FakeAppContext = ({
-  initialAppState = {},
-  children,
-}) => (
-  <AppContext.Provider value={initialAppState}>
+export const renderWithSearchContext = (children) => renderWithRouter(
```
Comment (raq929) [inline src/utils/tests.jsx:27]: Can you do me a favor and export this (from the package)? I want to use it in other repos. ;)

**Review by adamstankiewicz — APPROVED** [review]
LGTM. Curious about any findings you uncover about the other `history` related issue we chatted about :) But is not a blocker for this PR if that ends up taking longer than you're willing to give it right now.

**Pull request merged**